app/common/StringUnion.ts

```diff
@@ -24,11 +24,14 @@ export const StringUnion = <UnionType extends string>(...values: UnionType[]) =>
     return valueSet.has(value);
   };
 
+  let buildErrMessage = (actual: string, expected: string, _: string[]) => {
+    return `Value '${actual}' is not assignable to type '${expected}'.`;
+  };
   const check = (value: string): UnionType => {
     if (!guard(value)) {
       const actual = JSON.stringify(value);
       const expected = values.map(s => JSON.stringify(s)).join(' | ');
-      throw new TypeError(`Value '${actual}' is not assignable to type '${expected}'.`);
+      throw new TypeError(buildErrMessage(actual, expected, values));
```

Comment:
I prefer instead to define an error with more details, and let the caller rephrase:
```
class StringUnionError extends TypeError {
  constructor(errMessage: string, public readonly actual: string, public readonly values: string[]) {
    super(errMessage);
  }
}
```
This way errors from the same StringUnion type can be handled differently by different callers if needed.